packages/core/scripts/upgrade-blueprint-3.0.0-rename

```diff
@@ -0,0 +1,171 @@
+        shift
+        ;;
+        --prefix=*)
+        PREFIX="${i#*=}"
+        shift
+        ;;
+        -h|--help)
+        usage
+        exit 1
+        ;;
+        *)
+        ;;
+    esac
+done
+
+# Default argument values
+SEARCH_PATH=${SEARCH_PATH:-'.'}
+INCLUDE_GLOB=${INCLUDE_GLOB:-'*.{ts,tsx,scss,less}'}
+EXCLUDE_GLOB=${EXCLUDE_GLOB:-'{dist,build,node_modules}'}
+PREFIX=${PREFIX:-'(?!(?<=public )|(?<=protected )|(?<=abstract )|(?<=private )|(?<=this\.))'}
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+BLUE='\033[0;34m'
+RESET='\033[0m'
+
+function matchingFiles() {
+    # Find all files containing regexp. Use grep's glob syntax for
+    # include/exclude paths. Use perl's regexp syntax.
+    cmd="grep \
+        --recursive \
+        --files-with-matches \
+        --include=$INCLUDE_GLOB \
+        --exclude-dir=$EXCLUDE_GLOB \
+        . \
+        ${SEARCH_PATH} \
+        | xargs perl -lne 'if (/$1/) { print \$ARGV; close ARGV }'
+    "
+    echo "$(eval $cmd)"
+}
+
+function rename() {
+    # Name parameters
+    fromString=$1
+    toString=$2
+    findRegexp=${3:-$fromString}
+    files=$(matchingFiles "$findRegexp")
+
+    if [[ -z ${files// } ]]; then
+        echo -e "No files contain ${BLUE}${fromString}${RESET}"
+    else
+        count=$(echo "$files" | wc -l | awk '{print $1}')
+        echo -e "Renaming ${BLUE}${fromString}${RESET} -> ${GREEN}${toString}${RESET} in ${BLUE}${count}${RESET} files"
+
+        # Iterate and search&replace
+        echo "$files" | while read -r file ; do
+            echo -n "$file ... "
+
+            # Search&replace in place with perl
+            perl -p -i -e "s/$findRegexp/$toString/g" "$file"
+            echo -e "${GREEN}done${RESET}"
+        done
+    fi
+}
+
+function warn() {
+    # Name parameters
+    fromString=$1
+    toString=$2
+    findRegexp=${3:-$fromString}
+    files=$(matchingFiles "$findRegexp")
+
+    if [[ -z ${files// } ]]; then
+        echo -e "No files contain ${BLUE}${fromString}${RESET}"
+    else
+        count=$(echo "$files" | wc -l | awk '{print $1}')
+        echo ""
+        echo -e "${RED}WARNING: Skipping ${BLUE}${fromString}${RESET} -> ${GREEN}${toString}${RESET}"
+        echo -e "${RED}Can't safely change ${BLUE}${fromString}${RED}, you'll need to manually rename/remove this string.${RESET}"
+        echo -e "${RED}It was found in these files ${BLUE}${count}${RED} files:${RESET}"
+        echo "$files"
+    fi
+}
+
+function renameProp() {
+    # Add prefix and word boundaries to search string
+    rename $1 $2 "$PREFIX\\b$1\\b"
+}
+
+function warnProp() {
+    warn $1 $2 "$PREFIX\\b$1\\b"
+}
+
+function renamePartialClass() {
+    # Don't add word boundary so that partial css classnames rename
+    rename $1 $2 "\\b$1"
+}
+
+function renameString() {
+    # Add simple word boundaries
+    rename $1 $2 "\\b$1\\b"
+}
+
+echo "
+
+Renames for @blueprintjs/core"
+renameProp visual icon
+renameString "Classes\.CALLOUT_TITLE" "Classes.HEADING" 
```

Comment:
I didn't actually test these class renames, maybe I should before we merge?

Reply:
we can always do follow ups and stuff, but let's make sure it works before merging yes

Reply:
tested manually, seems to work correctly 👍